src/main.rs

```diff
@@ -277,6 +284,20 @@ enum SemverChecksCommands {
     CheckRelease(CheckRelease),
 }
 
+#[derive(Copy, Clone, ValueEnum, Debug)]
+enum LintLevel {
+    Major,
+    Minor,
```

Comment:
imo we should have a `Patch` level, even if we don't use it **yet** but I leave that to you all

Reply:
That doesn't make sense, what would a `Patch` level lint be? A `Minor` lit checks if the API changes require a minor version increment. There is nothing less than a `Patch` level increment; so checking for things that would require a `Patch` level increment. That's why `RequiredSemverUpdate` only has those two options as well.

I think there still is a misunderstanding / misconsensus what we are trying to achieve here. I think we are talking on different pages.

The essential question for me is: do you agree with [this statement](https://github.com/obi1kenobi/cargo-semver-checks/issues/359#issuecomment-1424966450)?

Reply:
@epage As I said earlier, there are two things that seem to cause confusion in the discussions:
- Overriding the version difference vs additional filtering after the version check
- The confusing correspondence between filter levels, lint levels and version increment levels.
  - `Minor` lints check if a `minor` version increment is required. Those checks should be silenced if a `minor` change was already detected
    - `--lint-level major` means that only `major` tests should run, meaning we only check whether or not we need a `major` version increment. Therefore this flag the same effect as a `minor` version increment (this is the part that confuses people, I think; but without looking at the implementation details, it's the most intuitive way)
  - `Major` lints check if a `major` version increment is required; Those checks should be silenced if a `major` change was detected

Maybe this is more intuitive:
   - `--lint-level minor` checks for both `major` and `minor` lints and fails if the new version in `Cargo.toml` does not match those changes
   - `--lint-level major` only checks for `major` lints and ignores `minor` lints. It also only fails if the version in `Cargo.toml` does not match those changes.

Reply:
Alternatively, we could introduce a flag `--skip-minor-lints` that would do the same thing as `--lint-level major`. Maybe that would be more intuitive.